components/raftstore/src/store/fsm/peer.rs

```diff
@@ -3481,6 +3529,22 @@ where
                             .raft_group
                             .raft
                             .adjust_max_inflight_msgs(peer_id, self.ctx.cfg.raft_max_inflight_msgs);
+                        // TODO: if witness -> non-witness
```

Comment:
This shouldn't be coupled with witness feature, no matter whether it has a witness, it needs trace apply index

Reply:
e.g. pre-transfer leader check can leverage it as well.

Reply:
Got it, thanks!